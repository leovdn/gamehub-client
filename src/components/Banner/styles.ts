import styled, { css } from 'styled-components'

export const Wrapper = styled.main`
  position: relative;

  @media (min-width: 768px) {
    box-shadow: 0 0.4rem 0.5rem 0 rgba(0, 0, 0, 0.2);
  }
`

type ImgProps = {
  src: string
}

export const Img = styled.div<ImgProps>`
  ${({ theme, src }) => css`
    width: 100%;
    height: 23rem;
    background-color: ${theme.colors.lightGray};
    background-image: url(${src});
    background-position: center center;
    background-size: cover;

    @media (min-width: 768px) {
      height: 58rem;
    }
  `}
`

export const Caption = styled.div`
  ${({ theme }) => css`
    width: 100%;
    background-color: rgba(0, 0, 0, 0.7);
    padding: ${theme.spacings.small};

    @media (min-width: 768px) {
      border-radius: 0 0 ${theme.border.radius} ${theme.border.radius};
      padding: ${theme.spacings.large};
      position: absolute;
      bottom: 0;
      left: 0;
    }
  `}
`

export const Title = styled.h2`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.large};
    font-weight: ${theme.font.bold};
    color: ${theme.colors.white};

    @media (min-width: 768px) {
      font-size: ${theme.font.sizes.xxlarge};
    }
  `}
`

export const Subtitle = styled.h3`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    font-weight: ${theme.font.light};
    color: ${theme.colors.white};
    margin-bottom: ${theme.spacings.medium};

    strong {
      font-weight: ${theme.font.bold};
      color: ${theme.colors.primary};
    }

    @media (min-width: 768px) {
      font-size: ${theme.font.sizes.xlarge};
    }
  `}
`
