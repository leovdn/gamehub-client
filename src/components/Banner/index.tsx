import Button from 'components/Button'
import * as S from './styles'
import Ribbon from 'components/Ribbon'

export type BannerProps = {
  img: string
  title: string
  subtitle: string
  buttonLabel: string
  buttonLink: string
  ribbon?: string
  ribbonColor?: 'primary' | 'secondary'
  ribbonSize?: 'normal' | 'small'
}

const Banner = ({
  img,
  title,
  subtitle,
  buttonLabel,
  buttonLink,
  ribbon,
  ribbonColor = 'primary',
  ribbonSize = 'normal'
}: BannerProps) => {
  return (
    <S.Wrapper>
      {ribbon && (
        <Ribbon color={ribbonColor} size={ribbonSize}>
          {ribbon}
        </Ribbon>
      )}

      <S.Img src={img} role="img" aria-label={title} />
      <S.Caption>
        <S.Title>{title}</S.Title>
        <S.Subtitle dangerouslySetInnerHTML={{ __html: subtitle }} />
        <Button as="a" href={buttonLink} size="large">
          {buttonLabel}
        </Button>
      </S.Caption>
    </S.Wrapper>
  )
}

export default Banner
