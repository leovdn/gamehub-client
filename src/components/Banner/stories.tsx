import Banner, { BannerProps } from '.'
import { Meta, StoryFn } from '@storybook/react'

export default {
  title: 'Banner',
  component: Banner,
  args: {
    img: 'https://images.unsplash.com/photo-1550745165-9bc0b252726f?q=80&w=1470&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D',
    title: 'Banner Default',
    subtitle: '<p>Play the new <strong>Game</strong> season</p>',
    buttonLabel: 'Buy now',
    buttonLink: '/games/default'
  },
  parameters: {
    layout: 'fullscreen'
  }
} as Meta

export const Default: StoryFn<BannerProps> = (args) => <Banner {...args} />
