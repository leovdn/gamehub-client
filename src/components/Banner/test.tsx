import { screen } from '@testing-library/react'

import Banner from './index'
import { renderWithTheme } from 'utils/tests/helpers'

const buttonProps = {
  img: 'https://images.unsplash.com/photo-1550745165-9bc0b252726f?q=80&w=1470&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D',
  title: 'Banner Default',
  subtitle: '<p>Play the new <strong>Game</strong> season</p>',
  buttonLabel: 'Buy now',
  buttonLink: '/games/default'
}

describe('<Banner />', () => {
  it('should render correctly', () => {
    const { container } = renderWithTheme(<Banner {...buttonProps} />)

    expect(
      screen.getByRole('heading', { name: /Banner Default/i })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('heading', { name: /Play the new Game season/i })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('img', { name: /Banner Default/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the Ribbon', () => {
    renderWithTheme(
      <Banner
        {...buttonProps}
        ribbon="My Ribbon"
        ribbonSize="small"
        ribbonColor="secondary"
      />
    )

    const ribbon = screen.getByText(/My Ribbon/i)
    expect(ribbon).toBeInTheDocument()
    expect(ribbon).toHaveStyle({ backgroundColor: '#3CD3C1' })
    expect(ribbon).toHaveStyle({ height: '2.6rem', fontSize: '1.2rem' })
  })
})
