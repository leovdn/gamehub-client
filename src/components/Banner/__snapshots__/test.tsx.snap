// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner /> should render correctly 1`] = `
.c5 {
  display: inline-flex;
  align-items: center;
  justify-content: center;
  padding: 0.8rem;
  border-radius: 0.4rem;
  border: none;
  background: linear-gradient(180deg,#ff4f5f 0%,#f062c0 50%);
  color: #FAFAFA;
  text-decoration: none;
  cursor: pointer;
  height: 5rem;
  font-size: 1.6rem;
  padding: 0.8rem 4.8rem;
}

.c5:hover {
  background: linear-gradient(180deg,#e35565 0%,#d958a6 50%);
}

.c0 {
  position: relative;
}

.c1 {
  width: 100%;
  height: 23rem;
  background-color: #EAEAEA;
  background-image: url(https://images.unsplash.com/photo-1550745165-9bc0b252726f?q=80&w=1470&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D);
  background-position: center center;
  background-size: cover;
}

.c2 {
  width: 100%;
  background-color: rgba(0,0,0,0.7);
  padding: 2.4rem;
}

.c3 {
  font-size: 1.8rem;
  font-weight: 600;
  color: #FAFAFA;
}

.c4 {
  font-size: 1.4rem;
  font-weight: 300;
  color: #FAFAFA;
  margin-bottom: 3.2rem;
}

.c4 strong {
  font-weight: 600;
  color: #F231A5;
}

@media (max-width:768px) {

}

@media (min-width:768px) {
  .c0 {
    box-shadow: 0 0.4rem 0.5rem 0 rgba(0,0,0,0.2);
  }
}

@media (min-width:768px) {
  .c1 {
    height: 58rem;
  }
}

@media (min-width:768px) {
  .c2 {
    border-radius: 0 0 0.4rem 0.4rem;
    padding: 4.0rem;
    position: absolute;
    bottom: 0;
    left: 0;
  }
}

@media (min-width:768px) {
  .c3 {
    font-size: 2.8rem;
  }
}

@media (min-width:768px) {
  .c4 {
    font-size: 2.0rem;
  }
}

<main
  class="c0"
>
  <div
    aria-label="Banner Default"
    class="c1"
    role="img"
    src="https://images.unsplash.com/photo-1550745165-9bc0b252726f?q=80&w=1470&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D"
  />
  <div
    class="c2"
  >
    <h2
      class="c3"
    >
      Banner Default
    </h2>
    <h3
      class="c4"
    >
      <p>
        Play the new 
        <strong>
          Game
        </strong>
         season
      </p>
    </h3>
    <a
      class="c5"
      href="/games/default"
    >
      <span>
        Buy now
      </span>
    </a>
  </div>
</main>
`;
